irs/src/components/Navigation/MainNav.js

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import {
+  Collapse,
+  Navbar,
+  NavbarToggler,
+  NavbarBrand,
+  Nav,
+  NavItem,
+  NavLink,} from 'reactstrap';
+
+class MainNav extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.toggle = this.toggle.bind(this);
+    this.state = {
+      isOpen: false
+    };
+  }
+  toggle() {
+    this.setState({
+      isOpen: !this.state.isOpen
+    });
+  }
```

Comment:
since you reference the current `state` inside your setState method here, it might cause errors while being batched for the update -- instead, use the `prevState` pattern where you pass in the previous version of the state object:
`this.setState(prevState => {isOpen: !prevState.isOpen})`